lib/src/main/java/io/ably/lib/types/BaseMessage.java

```diff
@@ -93,9 +98,13 @@ public void decode(ChannelOptions opts) throws AblyException {
 						catch(JsonParseException e) { throw AblyException.fromThrowable(e); }
 						continue;
 					}
-					if(xform == "cipher" && opts != null && opts.encrypted) {
-						data = opts.getCipher().decrypt((byte[])data);
-						continue;
+					if(xform == "cipher") {
+						if (opts != null && opts.encrypted) {
+							data = opts.getCipher().decrypt((byte[]) data);
+							continue;
+						}
+						else
+							Log.e(TAG, "Encrypted message received but encryption is not set up");
```

Comment:
I would if it was easy to get channel name at this point. Most I can do is throw custom AblyException, catch it in Channel.onMessage and log it there. But it seems like a lot of effort for a minor thing